### DISCUSSION TITLE
Renovate not picking up Docker image

### DISCUSSION THREAD

**Discussion by syphernl** [renovatebot/renovate#9332] [Request Help] [answered]
We have Dockerfile of which the updates are not being picked up by Renovate. 

This is our Dockerfile:

```dockerfile
# renovate: datasource=docker depName=zabbix/zabbix-web-nginx-pgsql
ARG ZABBIX_VERSION=5.2.5

FROM zabbix/zabbix-web-nginx-pgsql:${ZABBIX_VERSION}-alpine as app

.... some extra stuff ...
```

The approach is the same as being used in [docker-renovate-full](https://github.com/renovatebot/docker-renovate-full/blob/main/Dockerfile#L1)

Currently `5.2.6` is available, yet Renovate doesn't make an MR with the change and the `ZABBIX_VERSION` stays at `5.2.5`.

We based our regexManager on that mentioned in the [renovate docs](https://docs.renovatebot.com/modules/manager/regex/#advanced-capture):

```js
  regexManagers: [
    {
      fileMatch: ['^Dockerfile$'],
      matchStrings: [
        // Deal with ENV BLAH_VERSION entries using a depName
        'datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s',

        // Deal with ENV BLAH_VERSION entries using a lookupName
        'datasource=(?<datasource>.*?) lookupName=(?<lookupName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s',

        // Deal with ARG VERSION entry (used in geodesic container)
        'datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sARG VERSION=(?<currentValue>.*)\\s',

        // Deal with ARG BLAH_VERSION entries using a depName
        'datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sARG .*?_VERSION=(?<currentValue>.*)\\s',
      ],
      versioningTemplate: '{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}',
    },
  ],
```

The 1st and 3rd rules work fine on other Dockerfiles. In this particular case the 4th rule should match with our Dockerfile yet it doesn't. 

I have tested splitting up the regexManagers to only contain the 4th matchString, no luck.

I have also tried this approach (although I rather do it using the renovate comments):

```js
    {
      fileMatch: ['^Dockerfile$'],
      matchStrings: ['ARG ZABBIX_VERSION=(?<currentValue>.*)'],
      datasourceTemplate: 'docker',
    },
```
Which doesn't work either.

What am I doing wrong here? 

Logs:
```text
TRACE: Merged config.regexManagers
       "result": [
         {
           "fileMatch": ["^Dockerfile$"],
           "matchStrings": [
             "datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s",
             "datasource=(?<datasource>.*?) lookupName=(?<lookupName>.*?)( versioning=(?<versioning>.*?))?\\sENV .*?_VERSION=(?<currentValue>.*)\\s",
             "datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sARG VERSION=(?<currentValue>.*)\\s",
             "datasource=(?<datasource>.*?) depName=(?<depName>.*?)( versioning=(?<versioning>.*?))?\\sARG .*?_VERSION=(?<currentValue>.*)\\s"
           ],
           "versioningTemplate": "{{#if versioning}}{{{versioning}}}{{else}}semver{{/if}}"
         },
         {
           "fileMatch": ["^Dockerfile$"],
           "matchStrings": ["ARG ZABBIX_VERSION=(?<currentValue>.*)"],
           "datasourceTemplate": "docker"
         }
       ]
 WARN: Configuring logLevel in CLI or file is deprecated. Use LOG_LEVEL environment variable instead
 INFO: Repository started (repository=REPO/zabbix-frontend)
       "renovateVersion": "24.98.1"
 INFO: Dependency extraction complete (repository=REPO/zabbix-frontend)
       "baseBranch": "master",
       "stats": {
         "managers": {
           "dockerfile": {"fileCount": 1, "depCount": 1},
           "gitlabci-include": {"fileCount": 1, "depCount": 5},
           "regex": {"fileCount": 2, "depCount": 2}
         },
         "total": {"fileCount": 4, "depCount": 8}
       }
 INFO: Repository finished (repository=REPO/zabbix-frontend)
       "durationMs": 4946

```

**Comment by viceice** [comment]
please create a minimal reporduction and provide the debug logs

**Reply by syphernl** [reply]
Logs have now been added

**Reply by rarkins** [reply]
This is definitely too complex to troubleshoot without a public reproduction. You could consider having each `matchStrings` in separate regex manager objects just in case it's the combining of them which doesn't work. Then if they do still (partially) work as now, reduce your reproduction until it's just the line and regex manager which doesn't work

**Comment by syphernl** [comment, marked as answer]
I have also tried it using the [approach the renovate repo's are using](https://github.com/renovatebot/.github/blob/main/default.json#L106), but this didn't pick it up either when using the `docker` datasource. Switching to `github-tags` seems to be working though.

My `Dockerfile` now looks like this:
```docker
# renovate: datasource=github-tags depName=zabbix/zabbix-docker
ARG ZABBIX_VERSION=5.2.5

FROM zabbix/zabbix-server-pgsql:${ZABBIX_VERSION}-alpine as app
```

Renovate now properly sees the `5.2.6` release and creates an MR for that. As an added benefit it also shows the releasenotes.

**Reply by rarkins** [reply]
Do you mean it's working now? Is there anything left to troubleshoot?

**Reply by syphernl** [reply]
It _is_ working when I use `datasource=github-tags`, it isn't when `datasource=docker` is being used. Unfortunately even in `LOG_MODE=trace` you won't see what type of requests Renovate does to external sites, so I can't tell whether it's looking using the wrong input parameters for instance.

**Reply by rarkins** [reply]
What does the "packageFiles with updates" log message say about it when you use docker datasource? e.g. is it detected, and what's the object that's printed for it?
